<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainwindow</class>
 <widget class="QWidget" name="mainwindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1019</width>
    <height>809</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>mainwindow</string>
  </property>
  <widget class="QTreeWidget" name="treeWidget">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>0</y>
     <width>351</width>
     <height>441</height>
    </rect>
   </property>
   <column>
    <property name="text">
     <string notr="true">1</string>
    </property>
   </column>
  </widget>
  <widget class="QTreeWidget" name="treeWidget_2">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>0</y>
     <width>351</width>
     <height>441</height>
    </rect>
   </property>
   <column>
    <property name="text">
     <string notr="true">1</string>
    </property>
   </column>
  </widget>
  <widget class="QPlainTextEdit" name="plainTextEdit">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="plainTextEdit_2">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>20</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="insertApushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>70</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Insert</string>
   </property>
  </widget>
  <widget class="QPushButton" name="removeApushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>120</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Remove</string>
   </property>
  </widget>
  <widget class="QPushButton" name="findApushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>170</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Find</string>
   </property>
  </widget>
  <widget class="QPushButton" name="traverse1pushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>450</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>прямой обход</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="textEdit">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>450</y>
     <width>351</width>
     <height>351</height>
    </rect>
   </property>
  </widget>
  <widget class="QTextEdit" name="textEdit_2">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>450</y>
     <width>351</width>
     <height>351</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="traverse2pushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>500</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>обратный обход</string>
   </property>
  </widget>
  <widget class="QPushButton" name="traverse3pushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>550</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>по возрастанию
ключа</string>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="plainTextEdit_3">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>230</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="subtreeApushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>280</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>удалить
поддерево</string>
   </property>
  </widget>
  <widget class="QPushButton" name="ltreeApushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>330</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>удалить
левую ветвь</string>
   </property>
  </widget>
  <widget class="QPushButton" name="rtreeApushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>380</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>удалить
правую ветвь</string>
   </property>
  </widget>
  <widget class="QPushButton" name="traverse5pushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>510</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>обратный обход</string>
   </property>
  </widget>
  <widget class="QPushButton" name="traverse6pushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>560</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>по возрастанию
ключа</string>
   </property>
  </widget>
  <widget class="QPushButton" name="traverse4pushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>460</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>прямой обход</string>
   </property>
  </widget>
  <widget class="QPushButton" name="ltreeBpushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>330</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>вставить
левую ветвь</string>
   </property>
  </widget>
  <widget class="QPushButton" name="subtreeBpushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>280</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>вставить
поддерево</string>
   </property>
  </widget>
  <widget class="QPushButton" name="rtreeBpushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>380</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>вставить
правую ветвь</string>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="plainTextEdit_4">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>230</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="insertBpushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>70</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Insert</string>
   </property>
  </widget>
  <widget class="QPushButton" name="removeBpushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>120</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Remove</string>
   </property>
  </widget>
  <widget class="QPushButton" name="findBpushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>170</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>Find</string>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="xplainTextEdit">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>630</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="yplainTextEdit">
   <property name="geometry">
    <rect>
     <x>590</x>
     <y>630</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="lcapushButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>680</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>найти наименьшего
общего предка</string>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="plainTextEdit_5">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>730</y>
     <width>131</width>
     <height>41</height>
    </rect>
   </property>
   <property name="textInteractionFlags">
    <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
